core-java-modules/core-java-date-operations-4/src/test/java/com/baeldung/timestamptocalendar/SqlTimestampToCalendarConverterUnitTest.java

```diff
@@ -0,0 +1,37 @@
+package com.baeldung.timestamptocalendar;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+import java.sql.Timestamp;
+import java.util.Calendar;
+
+import org.junit.Test;
+
+public class SqlTimestampToCalendarConverterUnitTest {
+
+    @Test
+    public void givenTimestamp_whenConvert_thenEqualMillis() {
+        Timestamp timestamp = new Timestamp(124, 3, 19, 9, 30, 0, 801789562);
```

Comment:
This Timestamp constructor has been deprecated for a very long time (since Java 1.2 AFAIK) - we should really only be using the new Timestamp(long time) constructor + Timestamp.setNanos if you want to control the nanos field